pkg/opensearch/ppl_response_parser.go

```diff
@@ -86,12 +99,16 @@ func (rp *pplResponseParser) parseLogs(queryRes *backend.DataResponse, configure
 			doc[field.Name] = value
 		}
 
-		if configuredFields.LogLevelField != "" {
+		if isLogsQuery && configuredFields.LogLevelField != "" {
 			doc["level"] = doc[configuredFields.LogLevelField]
 		}
 
 		doc = flatten(doc, maxFlattenDepth)
 		for key := range doc {
+			// Do not add _source field (besides logs) as we are showing each _source field in table instead
```

Comment:
I copied this comment from the frontend code, honestly it's not that clear to me, I can update it if someone has a better understanding of what it means.

Reply:
my understanding is that `_source` is the original payload that elasticsearch/opensearch indexes.

e.g.

```json
{
  "src": "CA",
  "dst": "US",
  "start_date": "2023-10-25",
  "end_date": "2023-10-26",
  "geo": {
    "lat": 49.00000,
    "lon": 10.00000
  }
}
```

when you use the logs panel to display a logs query, i think it treats the `_source` field as the log message field (the field that is the original payload that the data is derived from) unless you specify another field.

for table formatted queries, you probably wouldn't want a column that just displays the json of the original payload since each of the fields in the original payload is already its own column.